Packs/Code42/Integrations/Code42/Code42.py

```diff
@@ -15,509 +17,650 @@
-    'Presentation': 'PRESENTATION',
-    'Script': 'SCRIPT',
-    'Spreadsheet': 'SPREADSHEET',
-    'Video': 'VIDEO',
-    'VirtualDiskImage': 'VIRTUAL_DISK_IMAGE',
-    'Archive': 'ARCHIVE'
+    "SourceCode": "SOURCE_CODE",
+    "Audio": "AUDIO",
+    "Executable": "EXECUTABLE",
+    "Document": "DOCUMENT",
+    "Image": "IMAGE",
+    "PDF": "PDF",
+    "Presentation": "PRESENTATION",
+    "Script": "SCRIPT",
+    "Spreadsheet": "SPREADSHEET",
+    "Video": "VIDEO",
+    "VirtualDiskImage": "VIRTUAL_DISK_IMAGE",
+    "Archive": "ARCHIVE",
 }
 
-SECURITY_EVENT_HEADERS = ['EventType', 'FileName', 'FileSize', 'FileHostname', 'FileOwner', 'FileCategory', 'DeviceUsername']
-SECURITY_ALERT_HEADERS = ['Type', 'Occurred', 'Username', 'Name', 'Description', 'State', 'ID']
+SECURITY_EVENT_HEADERS = [
+    "EventType",
+    "FileName",
+    "FileSize",
+    "FileHostname",
+    "FileOwner",
+    "FileCategory",
+    "DeviceUsername",
+]
+SECURITY_ALERT_HEADERS = ["Type", "Occurred", "Username", "Name", "Description", "State", "ID"]
+
+
+def _create_alert_query(event_severity_filter, start_time):
+    alert_filters = []
+    # Create alert filter
+    if event_severity_filter:
+        f = event_severity_filter
+        severity_filter = (f.upper() if isinstance(f, str) else list(map(lambda x: x.upper(), f)))
+        alert_filters.append(Severity.is_in(severity_filter))
+    alert_filters.append(AlertState.eq(AlertState.OPEN))
+    alert_filters.append(DateObserved.on_or_after(start_time))
+    alert_query = AlertQuery(*alert_filters)
+    alert_query.sort_direction = "asc"
+    return alert_query
 
 
 class Code42Client(BaseClient):
     """
-    Client will implement the service API, should not contain Demisto logic.
+    Client will implement the service API, should not contain Cortex XSOAR logic.
     Should do requests and return data
     """
 
     def __init__(self, sdk, base_url, auth, verify=True, proxy=False):
         super().__init__(base_url, verify=verify, proxy=proxy)
-        # Create the Code42 SDK instnace
-        self._sdk = sdk.create_using_local_account(base_url, auth[0], auth[1])
-        py42.settings.set_user_agent_suffix("Demisto")
+        # Create the Code42 SDK instance
+        self._sdk = sdk or py42.sdk.from_local_account(base_url, auth[0], auth[1])
+        py42.settings.set_user_agent_suffix("Cortex XSOAR")
 
-    def add_user_to_departing_employee(self, username, departure_epoch=None, note=None):
-        de = self._sdk.employee_case_management.departing_employee
+    def add_user_to_departing_employee(self, username, departure_date=None, note=None):
         try:
-            res = de.create_departing_employee(username, departure_epoch=departure_epoch, notes=note)
+            user_id = self.get_user_id(username)
+            self._sdk.detectionlists.departing_employee.add(
+                user_id, departure_date=departure_date
+            )
+            not note or self._sdk.detectionlists.update_user_notes(note)
```

Comment:
It's weird how their linter didn't complain about this one. At first I thought it was because `update_user_notes()` returns a `Py42Response` but it turns out that it doesn't actually.

Reply:
ugh, it definitely _should_ return a `Py42Response`. I just looked and there are a bunch of things in that client that don't. I'll file a ticket.